### ISSUE TITLE
Frontend edit of DateTimeField fails with ValidationError "Enter a valid date/time."

### ISSUE THREAD

**Issue by christianwgd** [stephenmcd/mezzanine#1838]
* Mezzanine 4.2.3 (also tested development)
* Django 1.8 to 1.11 (no error with 1.8, error with > 1.8)
* Python 2.7.12  
* SQLite 3.8.11  
* Windows 7  (also tested on Macos and Ubuntu)

Steps to reproduce:
- Fresh install venv with mezzanine, mezzanine-project projectname
- create Blogentry, try to change publish_date from frontend

Tried Django versions 1.9 to 1.11 with no luck. Debugging shows that the DateTime formfield gets a list with date and time, but requires a single string with date and time. 

At last tried Django 1.8 and recognized a deprecation warning saying 
\lib\site-packages\django\forms\fields.py:162: RemovedInDjango19Warning: Using SplitDateTimeWidget with DateTimeField is deprecated. Use SplitDateTimeField instead.
  value = self.to_python(value)

I made a few attempts to change the formfield class of the form DateTimeField to SplitDateTimeField but with no success. Changing the form class in mezzanine/forms/fields.py seems to have no effect.

**Comment by jerivas** [comment]
Hello!

I think this is Django issue. The date/time picker is a convenience offered in the admin, but if you're editing on the frontend you'll need to provide your own JavaScript to coerce whatever the user enters into the string expected by Django.

Feel free to re-open if I didn't understand correctly.

**Comment by christianwgd** [comment]
I'm a little bit confused. The fronted editing is part of the blog app. And it worked correct up to django 1.8, so i suppose it should work with newer versions of django. Django made a braking change to the mezzanine blog frontend editing by removing the support of SplitDateTimeWidget in conjunction with a DateTimeField. They recommended to use SplitDateTimeField instead. I think this change was never made in mezzanine. I don't think we should require people to create custom javascript code when setting up a mezzanine instance, only to make the blog app work correctly.
I appreciate the software quality of mezzanine, so let's keep this issue open until anyone in the community (including myself) had the chance to provide a solution for this bug. I did not have time to dig deep enough into the mezzanine code to find the right place to fix this, but had not given up yet.

**Comment by christianwgd** [comment]
I'm not able to reopen, since you, jerivas, closed this as a collaborator, sorry. Thanks for reopen it.

**Comment by jerivas** [comment]
Okay, I tried frontend editing the `publish_date` of a BlogPost (Django 1.11 + Mezzanine master and Django 1.8 + Mezzanine 4.2.3) and it is working fine. I get a regular text input, but as long as I enter the correctly-formatted string everything works as expected.

Could you elaborate a bit more on what you're experiencing?

**Comment by christianwgd** [comment]
This is what I entered:
<img width="601" alt="datetime" src="https://user-images.githubusercontent.com/9281647/35767190-d8f201ae-08e7-11e8-9ea1-44fbed606064.png">
This is the error i get:
<img width="455" alt="error" src="https://user-images.githubusercontent.com/9281647/35767198-f41592d4-08e7-11e8-935c-2edf4090388a.png">

Does this look like yours? And do you get the deprecation warning i mentioned above when trying Django 1.8?

**Comment by christianwgd** [comment]
At least I was able to reach a state as described by jerivas. The change gives me a text input, where i can write in a corrctly formatted string. I changed the DATE_TIME Widget (forms/fields.py:74) to "forms.DateTimeInput" and the style of the input field width to 150px (core/static/mezzanine/css/editable.css).
![datetimeinput](https://user-images.githubusercontent.com/9281647/36152946-fd9412e4-10cc-11e8-9411-775f227bc2df.PNG)
(this time using german localization)
No dropdowns anymore but working.
